src/main/java/randoop/contract/ObserverEqArray.java

```diff
@@ -0,0 +1,217 @@
+   * e.g. prints the second parameter in assertArrayEquals(var, new int[] {1,2,3})
+   *
+   * @return String that represents an instantiation of an array equal to value
+   */
+  private String printArray() {
+    String finalString = "";
+    if (value == null) {
+      finalString += "null";
+    } else {
+      finalString += "new ";
+      finalString += value.getClass().getCanonicalName();
+      finalString += printArrayComponents();
+    }
+    return finalString;
+  }
+
+  /**
+   * Converts an Object into its array representation
+   *
+   * @param value - the given Object
+   * @return its array representation
+   */
+  private Object[] toObjectArray(Object value) {
+    if (value instanceof byte[]) {
+      return ArrayUtils.toObject((byte[]) value);
+    } else if (value instanceof short[]) {
+      return ArrayUtils.toObject((short[]) value);
+    } else if (value instanceof int[]) {
+      return ArrayUtils.toObject((int[]) value);
+    } else if (value instanceof long[]) {
+      return ArrayUtils.toObject((long[]) value);
+    } else if (value instanceof float[]) {
+      return ArrayUtils.toObject((float[]) value);
+    } else if (value instanceof double[]) {
+      return ArrayUtils.toObject((double[]) value);
+    } else if (value instanceof char[]) {
+      return ArrayUtils.toObject((char[]) value);
+    } else if (value instanceof boolean[]) {
+      return ArrayUtils.toObject((boolean[]) value);
+    } else {
+      return (Object[]) value;
+    }
+  }
+
+  /**
+   * Helper method that prints the components of the array
+   *
+   * @return String that represents the components of the array
+   */
+  private String printArrayComponents() {
+    String finalString = "{";
+    int length = Array.getLength(value);
+    if (value.getClass().getComponentType().isEnum()) {
+      for (int i = 0; i < length; i++) {
+        if (Array.get(value, i) == null) {
+          finalString += "null";
+        } else {
+          finalString += new EnumValue((Enum<?>) Array.get(value, i)).getValueName();
+        }
+        if (i < length - 1) {
+          finalString += ", ";
+        }
+      }
+    } else if (value.getClass().getComponentType() == Class.class) {
+      for (int i = 0; i < length; i++) {
+        if (Array.get(value, i) == null) {
+          finalString += "null";
+        } else {
+          finalString += ((Class<?>) Array.get(value, i)).getCanonicalName() + ".class";
+        }
+        if (i < length - 1) {
+          finalString += ", ";
+        }
+      }
+    } else {
+      for (int i = 0; i < length; i++) {
+        finalString += Value.toCodeString(Array.get(value, i));
+        if (i < length - 1) {
+          finalString += ", ";
+        }
+      }
+    }
+    finalString += "}";
+    return finalString;
+  }
+
+  @Override
+  public boolean evaluate(Object... objects) throws Throwable {
+    assert objects.length == 1;
+    return Arrays.equals(toObjectArray(value), toObjectArray(objects[0]));
```

Comment:
Should this use the JUnit methods?  This differs from the code as written, because it does not use a delta when comparing floating-point arrays (float and double).

Reply:
Do you mean that I should use assertArrayEquals? If so, should I return "true" or "false" depending on if an assertion exception is thrown? And should I change this for the equals() method too?

Reply:
Yes, use assertArrayEquals as you said in your comment.

Do not change the equals() method.  Doing so would make equals() non-transitive and violate the contract of equals().